### ISSUE TITLE
Adding HTTPStatusEvent Listener "Non-null type parameter" error

### ISSUE THREAD

**Issue by hqkirkland** [openfl/openfl#2113]
Hey all. I get this issue when setting an HTTPStatusEvent listener using the below code:

```
	public function login(username:String, password:String):Void
	{
		var loginVars:URLVariables = new URLVariables();
		loginVars.username = username;
		loginVars.password = password;
		
		var loginRequest:URLRequest = new URLRequest(Endpoints.SIGNIN);
		loginRequest.method = "POST";
		loginRequest.data = loginVars;
		
		var loginClient:URLLoader = new URLLoader();
		loginClient.addEventListener(HTTPStatusEvent.HTTP_RESPONSE_STATUS, restfulHandler);
		loginClient.load(loginRequest);
	}
```

Error: `[Fault] exception, information=TypeError: Error #2007: Parameter type must be non-null.`

If you set the event listener using the literal value instead of the field, `"httpResponseStatus"` it works just fine.

Target platform is Flash. Haven't tested on other platforms, will do if necessary.

**Comment by tour1st** [comment]
According to the Adobe Flash documentation, httpResponseStatus is only supported on AIR platform.
https://help.adobe.com/fr_FR/FlashPlatform/reference/actionscript/3/flash/net/URLLoader.html#event:httpResponseStatus

**Comment by hqkirkland** [comment]
Yes I’m sorry; you’re correct! Seems rather silly to make this AIR-only. Oh well..

- Hunter

On Dec 30, 2018, at 12:39 AM, tour1st <notifications@github.com<mailto:notifications@github.com>> wrote:


According to the Adobe Flash documentation, httpResponseStatus is only supported on AIR platform.
https://help.adobe.com/fr_FR/FlashPlatform/reference/actionscript/3/flash/net/URLLoader.html#event:httpResponseStatus

**Comment by MSGhero** [comment]
Well, AIR-only means that Flash Player can’t do it, as far as AS3. For Haxe, everything that was AIR-only could probably be implemented on every target besides Flash.
